### ISSUE TITLE
Error should occur at the timing when only defining the function which only returns `None`

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20250]
*Memo:
- mypy test.py
- mypy 1.18.2
- Python 3.14.0
- Windows 11

Printing `None` returned from the function which only returns `None` gets error as shown below:

```python
def func() -> None:
    return None

print(func())
# Error
```

> error: "func" does not return a value (it only ever returns None)

But, only calling the function which only returns `None` without printing `None` doesn't get error as shown below:

```python
def func() -> None:
    return None

func()
# No error
```

And, only defining the function which only returns `None` without calling the function doesn't get error as shown below:

```python
def func() -> None:
    return None
# No error
```

So, the error should occur at the timing when only defining the function which only returns `None` as shown below: 

```python
def func() -> None:
    return None
# Error
```

> error: "func" does not return a value (it only ever returns None)

**Comment by hauntsaninja** [comment]
This is the intended behaviour. If you don't think this warning has value, you can disable it with `--disable-error-code func-returns-value`. See also https://github.com/python/mypy/issues/3226
